dag_in_context/src/interval_analysis.egg

```diff
@@ -249,4 +265,73 @@
        (= (IntB hi) (hi-bound v))
       )
       ((set (hi-bound lhs) (IntB  hi)))
-      :ruleset interval-analysis)
\ No newline at end of file
+      :ruleset interval-analysis)
+
+
+
+; If the If takes a tuple
+(rule (
+        ; expr < value
+        (= pred (Bop (LessThan) expr value))
+        (= if_e (If pred inputs then else))
+        ; the left operand of the < is an input to the if region
+        (= expr (Get inputs i))
+        ; the right operand of the < has an upper bound
+        (= (IntB v) (hi-bound value))
+        ; context node inside the if region
+        (= ctx (InContext (InIf true pred inputs) (Arg ty)))
+        (HasType inputs ty)
+        ; usage of the value x inside the if region
+        (Get ctx i)
+      )
+      ((set (hi-bound (Get ctx i)) (IntB (- v 1)))) ; expr < value was true, so we know expr < (hi-bound value) - 1
```

Comment:
Why is there a `- 1` here? If value is 2 and we know expr<2, expr could still be 1, right?

Reply:
I think it's correct- the hi-bound is inclusive, so this code sets the hi-bound to 1.

Reply:
Then the code is right but the comment is wrong.